### DISCUSSION TITLE
Bengali translation

### DISCUSSION THREAD

**Discussion by mahmudunnabikajal** [vuejs-translations/guidelines#66] [Ideas]
### Bengali translation

---
> I have started translation in Bengali language. I think this will help make Vue.js easily adoptable by young Bengalis who are going to start their careers.
> 
> If you want to contribute, I'd be happy to welcome you on board ❤.
---
> বাংলা ভাষায় অনুবাদ শুরু করেছি। আমি মনে করি এটি Vue.js-কে বাঙালি তরুণদের জন্য সহজে গ্রহণযোগ্য করে তুলতে সাহায্য করবে যারা তাদের ক্যারিয়ার শুরু করতে যাচ্ছে।
> 
> আপনি যদি অবদান রাখতে চান, আমি আপনাকে বোর্ডে স্বাগত জানাতে পেরে খুশি হব ❤।
---

Repo: https://github.com/vuejs-translations/docs-bn
Site: https://bn.vuejs.org/

** Comment in this discussions if you want to contribute or have any suggestions**

**Comment by mahmudunnabikajal** [comment]
@yyx990803 @NataliaTepluhina @Jinjiang Would you please publish the Bangla translation on bn.vuejs.org? would love to increase its official status and possibly add more translators.

For the time being, those of us who adore the Vue ecosystem desire that the increasingly popular Bangla Docs version of VueJS, Nuxt, Router, and Pinia would be made available.


A big thank you!

**Reply by mahmudunnabikajal** [reply]
@yyx990803 @jay-es @Jinjiang @kazupon @kiaking @nazar1ua @wxsms 

https://github.com/mahmudunnabikajal/docs-bn
friendly reminder to transfer vue 3 Bangla docs in vue translations.

**Reply by nazar1ua** [reply]
@mahmudunnabikajal sorry, I'm not member of Vue team

**Reply by Jinjiang** [reply]
@mahmudunnabikajal I found you haven't detached your fork, which is required before transferring. See https://github.com/vuejs-translations/guidelines#high-level-steps-of-working-on-a-new-translation

Update: I sent you an invitation. After accept that you should be free to transfer the repo and maintain it in this org.

Cheers

/cc @yyx990803

**Reply by mahmudunnabikajal** [reply]
@Jinjiang Thanks for your reply. I accept your invitation and transfer my repo to vuejs-translations [doc-bn](https://github.com/vuejs-translations/docs-bn).

**issue:** unable to merge a pull request or directly change the main branch. 
![marge issue screenshot from pull request](https://img001.prntscr.com/file/img001/k7IrgHxvQ4a3Eh2DgJuukQ.png)

**Reply by Jinjiang** [reply]
@mahmudunnabikajal I reset your permission to that repo. Please try again and let me know if it still does not work. Thanks.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang I try different ways to change the code. But showing error:
 - branch is protected require pull a request.
 - after sending a pull request it shows "I don't have reviewed or write access. At least 1 approving review is required."

# Issue Reference:
on direct push to the main branch

![on direct push to the main branch](https://img001.prntscr.com/file/img001/SjkAJMzmQb6J3mGaoKSd9A.png)
---
on pull requests from this repo [#2](https://github.com/vuejs-translations/docs-bn/pull/2) and from the fork repo [#4](https://github.com/vuejs-translations/docs-bn/pull/4)

![on pull requests from this repo and from the fork repo](https://img001.prntscr.com/file/img001/oAgkO9JuRPG_Gog57hcFng.png)

**Reply by Jinjiang** [reply]
@mahmudunnabikajal I double checked. It might because the default repo settings like the main branch is protected as the screenshot:

<img width="852" alt="image" src="https://github.com/vuejs-translations/guidelines/assets/206848/37018a98-35f9-4156-adc0-be4ad9085b69">

<img width="597" alt="image" src="https://github.com/vuejs-translations/guidelines/assets/206848/ce2423ea-4b06-49da-857f-fcf27fd4752c">

I guess a walkaround is just approving the PR by yourself and then you are free to merge it.

On the other hand, I can help you to disable any of the protection options, however, those restrictions are all the best practices for git repo maintenance. Let me know if you need it.

Thanks.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang ok then remove this "require approvals" option. Because I am unable to marge my own pull request. Also delete "upstream" branch from protection rules.

# Reference: 

![brancprotectionrules](https://img001.prntscr.com/file/img001/wA665GXOQ4yMJXn8rlV7pQ.png)

**Reply by Jinjiang** [reply]
Done. And don't forget to detach the fork.
Cheers

**Reply by mahmudunnabikajal** [reply]
@Jinjiang  Everything is working perfectly and I message github support they said only the admin can request to detach the fork.

# Reference

![support response](https://github.com/vuejs-translations/guidelines/assets/60002195/e563cd86-1bf5-427d-b64e-9dda0815eb82)

**Reply by Jinjiang** [reply]
@mahmudunnabikajal I see. I created a ticket for that. Will keep updated with you. Thanks.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang "upstream" branch has to delete from protection rules.

# Reference
 - screenshot from https://github.com/vuejs-translations/guidelines/discussions/66#discussioncomment-6161515
![screenshot from https://github.com/vuejs-translations/guidelines/discussions/66#discussioncomment-6161515](https://img001.prntscr.com/file/img001/E0fy-JjlQfGcniB3WY-sIQ.png)

**Reply by Jinjiang** [reply]
done

**Reply by mahmudunnabikajal** [reply]
@Jinjiang Thanks ❤️. 
Is there any way to deploy the main branch in Netlify? Because after transferring the repository Netlify remove it from my account.

**Reply by Jinjiang** [reply]
@mahmudunnabikajal now you should already have the admin role. Feel free to re-do it on your own.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang
 demo site is live on https://docs-bn.netlify.app/ .

**Reply by mahmudunnabikajal** [reply]
@Jinjiang @yyx990803  Hello guys, our translation is done. We would love to go live officially on bn.vuejs.org .

**Reply by Jinjiang** [reply]
@mahmudunnabikajal great job! For the next step, feel free to create a PR to vuejs/docs like this https://github.com/vuejs/docs/pull/2448 then we will proceed. Thanks.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang I created a PR to vuejs/docs

Reference: [vuejs/docs#2508](https://github.com/vuejs/docs/pull/2508)

**Reply by Jinjiang** [reply]
@mahmudunnabikajal the new domain works now. One more issue: add a config file for vercel headers like this: https://github.com/vuejs-translations/docs-zh-cn/blob/main/vercel.json

**Reply by mahmudunnabikajal** [reply]
@Jinjiang ok.

**Reply by mahmudunnabikajal** [reply]
@Jinjiang config file for vercel headers added

Reference: https://github.com/vuejs-translations/docs-bn/blob/main/vercel.json

**Reply by Jinjiang** [reply]
Great. The PR has been merged. We will announce it via Twitter later. /cc @yyx990803

**Reply by mahmudunnabikajal** [reply]
@Jinjiang Thanks ❤️

**Comment by Muhammad-Sarfaraz** [comment]
@mahmudunnabikajal 

Hey there, I would like to contribute to the bengali Vuejs docs. As a native speaker of Bangla and I'm originally held from Bangladesh. I would be grateful if I could contribute to this project.

Thanks!

Previously I had Contributed to:
https://github.com/minbrowser/min
https://github.com/tighten/laravelversions

**Reply by mahmudunnabikajal** [reply]
Hey @Muhammad-Sarfaraz !
That sounds fantastic. Thank you so much for offering to contribute! We appreciate your willingness to make the documentation more accessible and helpful to the Bengali-speaking community. 

- Step 1: Make sure you read this [Vue Community Translation Guidelines](https://github.com/vuejs-translations/guidelines)
- Step 2: Fork this repo [docs-bn](https://github.com/vuejs-translations/docs-bn)
- Step 3: Here is the Translation Process vuejs-translations/docs-bn/issues/1 Choose 3/5 uncheck pages & comment on the issue to let others know.
- Step 4: Create a branch (ex: branch name = selected page name) for each page and translate
- Step 5: After completing the translation create a pull request to the main repo ```branch(branch example: api/general.md)```. Pull request pattern: ```docs: guide/extras/ways-of-using-vue.md translation```
- Step 6: Repeat this process

If you need any assistance or have any questions, feel free to mention here.

**Comment by rusulazomsumon** [comment]
Contribute করতে আগ্রহী

**Reply by mahmudunnabikajal** [reply]
@rusulazomsumon Thanks. You can check this [Instructions](https://github.com/vuejs-translations/docs-bn/wiki) page where you find all the necessary Translation Guide.

> [!IMPORTANT]
> Must read [Community Guidelines](https://github.com/vuejs-translations/guidelines)

**Comment by dhrupo** [comment]
I want to contribute to Bengali. I am a native speaker of Bengali and currently residing in Bangladesh.

**Reply by mahmudunnabikajal** [reply]
@dhrupo That's great. You can check this [Instructions](https://github.com/vuejs-translations/docs-bn/wiki) page where you find all the necessary Translation Guide.

> [!IMPORTANT]
> Must read [Community Guidelines](https://github.com/vuejs-translations/guidelines)

**Reply by dhrupo** [reply]
Thanks.
Is it okay if I work on partners.json file?

**Reply by mahmudunnabikajal** [reply]
Sorry. You can only change the `src` folder.

Check these issues for updates:
- [Recheck translation to solve issues](https://github.com/vuejs-translations/docs-bn/issues/39)
- [Follow-up of the translation process](https://github.com/vuejs-translations/docs-bn/issues/1)

[a comment by an automated account is not shown]

**Reply by mahmudunnabikajal** [reply]
@Rasel-bot thats great. You can check this [Instructions](https://github.com/vuejs-translations/docs-bn/wiki) page where you find all the necessary Translation Guidelines. Once complete create a pull request.

**Comment by msisaif** [comment]
আমি আগ্রহী

**Reply by mahmudunnabikajal** [reply]
@msisaif check the [Instructions page](https://github.com/vuejs-translations/docs-bn/wiki), where you'll discover all the essential Translation Guidelines. Afterward, feel free to create a pull request once you've completed any task.

**Comment by muhaiminShihab** [comment]
আমি বাংলা ডক্সকে বেটার করতে বাংলা ট্রান্সলেশন কন্ট্রীবিউট করতে আগ্রহী যাতে আমাদের আপকামিং ডেভেলপেররা মাতৃ ভাষায় খুব সহজে ভিউ জেএস শিখতে পারে।

**Reply by mahmudunnabikajal** [reply]
@muhaiminShihab  আপনি [নির্দেশনা](https://github.com/vuejs-translations/docs-bn/wiki) অনুযায়ী কান্ট্রিবিউশন শুরু করতে পারেন। এখানে আপনি সমস্ত প্রয়োজনীয় [অনুবাদ গাইড](https://github.com/vuejs-translations/docs-bn/wiki/Translation-Guide) পাবেন। যেকোনো সাপোর্ট এর জন্য [লিঙ্কেডিন](https://www.linkedin.com/in/mahmudun-nabi-kajal/) এ মেসেজ দিতে পারেন।

**Comment by 2u841r** [comment]
গুগল ট্রান্সলেট করা পুরাটাই! ভয়ঙ্কর

**Reply by mahmudunnabikajal** [reply]
@2u841r ডকুমেন্টেশন অনুবাদের সমস্যা প্রতিনিয়ত সমাধান করা হচ্ছে। আপনি অনুবাদের সমস্যাগুলি সমাধানে অবদান রাখতে পারেন৷

[Recheck Translation](https://github.com/vuejs-translations/docs-bn/issues/39)

**Comment by Meraj-Kazi** [comment]
Wanna start contributing on this. How to start? 
Thanks!

**Reply by mahmudunnabikajal** [reply]
@Meraj-Kazi You can start contributing according to [instructions](https://github.com/vuejs-translations/docs-bn/wiki). Here you will find all necessary [Translation Guide](https://github.com/vuejs-translations/docs-bn/wiki/Translation-Guide). For any support message [Linkedin](https://www.linkedin.com/in/mahmudun-nabi-kajal/).
